obspy/clients/neic/client.py

```diff
@@ -170,15 +172,30 @@ def get_waveforms_nscl(self, seedname, starttime, duration):
             (seedname, start, duration)
         if self.debug:
             print(ascdate() + " " + asctime() + " line=" + line)
+
+        # prepare for routing through http_proxy_connect
+        address = (self.host, self.port)
+        if self.proxy:
+            proxy = (self.proxy.hostname, self.proxy.port)
+            auth = ((self.proxy.username, self.proxy.password) if
+                    self.proxy.username else None)
+
         success = False
         while not success:
             try:
-                s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                if self.proxy:
+                    s, _, _ = http_proxy_connect(address, proxy, auth)
+                    # This socket is already connected to the proxy
+                else:
+                    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
                 with NamedTemporaryFile() as tf:
                     if self.debug:
                         print(ascdate(), asctime(), "connecting temp file",
                               tf.name)
-                    s.connect((self.host, self.port))
+                    if not self.proxy:
+                        # The proxy-ed socket is already connected
+                        s.connect((self.host, self.port))
```

Comment:
Does this really need to be in the `NamedTemporaryFile` context? Can it not just be in the `else` above?